Assets/MRTK/SDK/Experimental/Elastic/Scripts/ElasticsManager.cs

```diff
@@ -0,0 +1,222 @@
+            if (hostTransform != null)
+            {
+                if (elasticTypes.HasFlag(TransformFlags.Move))
+                {
+                    hostTransform.position = translationElastic.ComputeIteration(targetTransform.Position, Time.deltaTime);
+                }
+
+                if (elasticTypes.HasFlag(TransformFlags.Rotate))
+                {
+                    hostTransform.rotation = rotationElastic.ComputeIteration(targetTransform.Rotation, Time.deltaTime);
+                }
+
+                if (elasticTypes.HasFlag(TransformFlags.Scale))
+                {
+                    hostTransform.localScale = scaleElastic.ComputeIteration(targetTransform.Scale, Time.deltaTime);
+                }
+
+                return elasticTypes;
+            }
+            else 
+            {
+                return 0;
+            }
+        }
+
+        /// <summary>
+        /// Initialize elastics system with the given host transform.
+        /// Caches a reference to the host transform to be able to keep updating elastics after manipulation.
+        /// </summary>
+        /// <param name="elasticsTransform">host transform the elastics are applied to.</param>
+        public void InitializeElastics(Transform elasticsTransform)
+        {
+            hostTransform = elasticsTransform;
+            if (elasticTypes.HasFlag(TransformFlags.Move))
+            {
+                translationElastic = new VolumeElasticSystem(hostTransform.position,
+                                                             translationElastic?.GetCurrentVelocity() ?? Vector3.zero,
+                                                             translationElasticExtent,
+                                                             translationElasticConfigurationObject.ElasticProperties);
+            }
+            if (elasticTypes.HasFlag(TransformFlags.Rotate))
+            {
+                rotationElastic = new QuaternionElasticSystem(hostTransform.rotation,
+                                                              rotationElastic?.GetCurrentVelocity() ?? Quaternion.identity,
+                                                              rotationElasticExtent,
+                                                              rotationElasticConfigurationObject.ElasticProperties);
+            }
+            if (elasticTypes.HasFlag(TransformFlags.Scale))
+            {
+                scaleElastic = new VolumeElasticSystem(hostTransform.localScale,
+                                                       scaleElastic?.GetCurrentVelocity() ?? Vector3.zero,
+                                                       scaleElasticExtent,
+                                                       scaleElasticConfigurationObject.ElasticProperties);
+            }
+        }
+
+        #region MonoBehaviour Functions
+        private void Update()
```

Comment:
One other thought was around the per-frame computations here - once something has gotten the point where it's below the threshold, I'm curious if there's any way for us to set a bit to know to ignore it until the next call to ApplyTargetTransform. 

I don't know if this will really do that much (i.e. maybe the 3x magnitude calls vs 3 bool checks isn't too different in the long run).

Reply:
i'm caching the simulated elastics (flags) from prev frame now and according to that information I'm either not calling any of the simulation code or just the specific ones for the type that still needs simulating - also added all the exit criteria into a private method that gets evaluated before doing the squrmagnitude call.
that should hopefully cover everything :)